File: launchdarkly/ruby/src/CreateInsightGroupExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["ApiKey"] = "YOUR_API_KEY"
end

post_insight_group_params = LaunchDarklyClient::PostInsightGroupParams.new
post_insight_group_params.name = "Production - All Apps"
post_insight_group_params.key = "default-production-all-apps"
post_insight_group_params.project_key = "default"
post_insight_group_params.environment_key = "production"
post_insight_group_params.application_keys = [
    "billing-service",
    "inventory-service",
]

begin
    response = LaunchDarklyClient::InsightsScoresBetaApi.new.create_insight_group(
        post_insight_group_params,
    )

    p response
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling InsightsScoresBeta#create_insight_group: #{e}"
end

File: launchdarkly/python/src/CreateInsightGroupExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    post_insight_group_params = models.PostInsightGroupParams(
        name="Production - All Apps",
        key="default-production-all-apps",
        projectKey="default",
        environmentKey="production",
        applicationKeys=[
            "billing-service",
            "inventory-service",
        ],
    )

    try:
        response = api.InsightsScoresBetaApi(api_client).create_insight_group(
            post_insight_group_params=post_insight_group_params,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling InsightsScoresBetaApi#create_insight_group: %s\n" % e)
